src/snyk/cli/process.ts

```diff
@@ -1,38 +1,57 @@
-import { spawn } from 'child_process';
+import { ChildProcessWithoutNullStreams, spawn } from 'child_process';
 import { Configuration, IConfiguration } from '../common/configuration/configuration';
 import { ILog } from '../common/logger/interfaces';
 import { CLI_INTEGRATION_NAME } from './contants/integration';
+import { CliError } from './services/cliService';
 
 export class CliProcess {
   private readonly successExitCodes = [0, 1];
 
+  private runningProcess: ChildProcessWithoutNullStreams | null;
+
   constructor(private readonly logger: ILog, private readonly config: IConfiguration) {}
 
   /**
    * Returns CLI output given provided arguments.
    */
   spawn(cliPath: string, args: string[]): Promise<string> {
-    // todo: implement cancellation
     return new Promise((resolve, reject) => {
       let output = '';
 
-      const cli = spawn(cliPath, args, { env: this.getProcessEnv() });
+      this.runningProcess = spawn(cliPath, args, { env: this.getProcessEnv() });
 
-      cli.stdout.setEncoding('utf8');
-      cli.stdout.on('data', (data: string | Buffer) => (output += data));
+      this.runningProcess.stdout.setEncoding('utf8');
+      this.runningProcess.stdout.on('data', (data: string | Buffer) => (output += data));
 
-      cli.on('error', err => reject(err));
-      cli.on('close', (code, signal) => {
-        if (code && code in this.successExitCodes) {
+      this.runningProcess.on('error', err => {
+        this.runningProcess = null;
+        reject(err);
+      });
+      this.runningProcess.on('close', (code, signal) => {
+        if (code !== null && code in this.successExitCodes) {
+          this.runningProcess = null;
```

Comment:
shall we also turn `.off` the event listeners to avoid re-attaching them in the future?

Reply:
the listeners will be removed during garbage collection as `runningProcess` is set to null here, making it eligible for the GC. I've added the cleanup logic anyway to be safe on this matter in future 👍